src/main/java/seedu/teachstack/storage/JsonSerializableUserData.java

```diff
@@ -0,0 +1,71 @@
+package seedu.teachstack.storage;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
```

Comment:
Just want to confirm if we can use fasterxml. Otherwise, looks really good! The save functionality works as expected. If it can be used then I'll merge it in.

Reply:
Yes, it is also used in the base AB3 in all the json related storage classes.

Reply:
LGTM then. Thanks